File: solutions/typescript/resistor-color/1/resistor-color.ts
export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];

export const colorCode = (color: string) => {
  const id = COLORS.findIndex((c) => c === color);

  if (id > -1) {
    return id;
  }
  throw new Error("Delete this line and implement this function");
};

File: solutions/csharp/resistor-color/1/ResistorColor.cs
public static class ResistorColor
{
    public static int ColorCode(string color) => color switch
    {
        "black" => 0,
        "brown" => 1,
        "red" => 2,
        "orange" => 3,
        "yellow" => 4,
        "green" => 5,
        "blue" => 6,
        "violet" => 7,
        "grey" => 8,
        "white" => 9,
        _ => throw new ArgumentOutOfRangeException(nameof(color), color, null)
    };

    public static string[] Colors()
    {
        var colors = new string[10];
        
        return colors
            .Select((_, i) => i switch
            {
                0 => "black",
                1 => "brown",
                2 => "red",
                3 => "orange",
                4 => "yellow",
                5 => "green",
                6 => "blue",
                7 => "violet",
                8 => "grey",
                9 => "white",
                _ => throw new ArgumentOutOfRangeException(nameof(i), i, null)
            })
            .ToArray();
    }
}